flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/source/reader/deserializer/AvroToRowDataDeserializationSchema.java

```diff
@@ -22,25 +22,36 @@
 import org.apache.flink.table.data.RowData;
 import org.apache.flink.table.types.logical.RowType;
 
+import com.google.cloud.flink.bigquery.common.exceptions.BigQueryConnectorException;
 import org.apache.avro.generic.GenericRecord;
-
-import java.io.IOException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /** Simple implementation for the Deserialization schema (from Avro GenericRecord to RowData). */
 @Internal
 public class AvroToRowDataDeserializationSchema
         implements BigQueryDeserializationSchema<GenericRecord, RowData> {
     private final AvroToRowDataConverters.AvroToRowDataConverter converter;
     private final TypeInformation<RowData> typeInfo;
+    private static final Logger LOG =
+            LoggerFactory.getLogger(AvroToRowDataDeserializationSchema.class);
 
     public AvroToRowDataDeserializationSchema(RowType rowType, TypeInformation<RowData> typeInfo) {
         this.converter = AvroToRowDataConverters.createRowConverter(rowType);
         this.typeInfo = typeInfo;
     }
 
     @Override
-    public RowData deserialize(GenericRecord record) throws IOException {
-        return (GenericRowData) converter.convert(record);
+    public RowData deserialize(GenericRecord record) throws BigQueryConnectorException {
+        try {
+            return (GenericRowData) converter.convert(record);
+        } catch (RuntimeException e) {
+            LOG.error(
+                    String.format(
+                            "Error deserializing Avro Generic Record %s to Row Data.%nError: %s.%nCause:%s ",
```

Comment:
1. "Deserialize" is not the right word here. Try "convert" or something.
2. Why `RuntimeException`? If you want to catch a generic exception, then might as well capture `Exception`

Reply:
1. Changed
2. Changed to `Exception`